### PR TITLE
Changes to support use of TagHelpers in applications with precompiled

### PR DESCRIPTION
**Pull request by pranavkm**
views.

### CODE REVIEW

**Comment by dougbu** [comment]
looks like this may have an impact on tooling.  are they ready?

**Comment by dougbu** [comment]
:watch:

**Comment by pranavkm** [comment]
> looks like this may have an impact on tooling. are they ready? 

I'm not sure what this is about?

**Comment by dougbu** [comment]
@pranavkm tooling subclasses `ResolveDescriptorsInAssembly` in order to override `ResolveDescriptorsInAssembly()`.  they'll get a default `TagHelperTypeResolver` instance unless they make a change to call the other (new) `base` constructor.  another way to phrase my question is: is that instance the correct one for tooling scenarios?

File: src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorResolver.cs [hunk 1]
```diff
@@ -112,7 +117,8 @@ public IEnumerable<TagHelperDescriptor> Resolve([NotNull] TagHelperDescriptorRes
             var tagHelperTypes = _typeResolver.Resolve(assemblyName, documentLocation, errorSink);
 
             // Convert types to TagHelperDescriptors
-            var descriptors = tagHelperTypes.SelectMany(TagHelperDescriptorFactory.CreateDescriptors);
+            var descriptors = tagHelperTypes.SelectMany(
+                                        type => TagHelperDescriptorFactory.CreateDescriptors(assemblyName, type));
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorResolver.cs]: just hit enter for spacing, easier to maintain.

**Comment by pranavkm** [comment]
> is that instance the correct one for tooling scenarios?

Yes, this is specifically meant for precompilation - tooling can continue working with their current approach of using the default `TagHelperTypeResolver`. A change they could do is follow the same pattern that Mvc does - call the new ctor passing in their custom type resolver. This would avoid them having to specify a custom `ITagHelperDescriptorResolver`.

**Comment by dougbu** [comment]
:shipit:

**Pull request closed without merging**